### PR TITLE
Follow/Unfollow Button Animation

### PR DESCRIPTION
**Pull request by Fouad-sys**

### CODE REVIEW

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/social/FollowButton.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+/**
+ * An implementation of [ButtonColors] used to change the color of a button according to whether it
+ * is selected.
+ *
+ * @param selected a [State] of [Boolean] indicating whether the button is selected.
+ */
+private class FollowButtonColors(selected: State<Boolean>) : ButtonColors {
+
+  /** A [State] of [Boolean] indicating whether the button is selected. */
+  private val selected by selected
```
Comment (KurohanaJuri) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/social/FollowButton.kt:32]: Question : How can you assign a variable with the name ?

Comment (Fouad-sys) [reply mobile/src/main/java/ch/epfl/sdp/mobile/ui/social/FollowButton.kt:32]: Not sure I completely understood but the other selected name refers to the class parameter

**Review by alexandrepiveteau — APPROVED** [review]
LGTM, thanks !

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/social/FollowButton.kt (hunk 1 above)
Comment (alexandrepiveteau) [reply mobile/src/main/java/ch/epfl/sdp/mobile/ui/social/FollowButton.kt:32]: Indeed, this is a property delegate for the class parameter `selected: State<Boolean>`.

**Review by matt989253 — APPROVED** [review]
Nice, thanks! LGTM

**Pull request merged**